pmd-java/src/test/java/net/sourceforge/pmd/coverage/PMDCoverageTest.java

```diff
@@ -4,93 +4,84 @@
 
 package net.sourceforge.pmd.coverage;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.emptyString;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
 import static org.junit.jupiter.api.Assertions.fail;
 
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
-import java.util.Arrays;
-import java.util.List;
+import java.nio.file.Paths;
+import java.util.function.Consumer;
 
-import org.apache.commons.lang3.ArrayUtils;
-import org.apache.commons.lang3.StringUtils;
-import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.io.TempDir;
 
-import net.sourceforge.pmd.PMD;
+import net.sourceforge.pmd.PMDConfiguration;
+import net.sourceforge.pmd.PmdAnalysis;
 import net.sourceforge.pmd.internal.util.IOUtil;
 import net.sourceforge.pmd.lang.LanguageVersion;
 import net.sourceforge.pmd.lang.java.JavaLanguageModule;
 
 import com.github.stefanbirkner.systemlambda.SystemLambda;
 
-// TODO enable test
-@Disabled("Test is failing and was excluded on PMD 7 branch")
 class PMDCoverageTest {
```

Comment:
I'd argue this test should be removed altogether… we already have integration tests in `pmd-dist`, so this add no extra value…

Reply:
Well as per #4522 this test actually discovered a bug in the dataflow code, so I would argue it is useful. Also it is executed when building `pmd-java`. I rarely ever build pmd-dist to test a change in pmd-java